lighthouse-core/lib/errors.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const strings = require('./strings');
+
+class LighthouseError extends Error {
+  constructor(errorDefinition) {
+    super(errorDefinition.code);
```

Comment:
feels like we should also set `this.code` to be the code. that clean string should will persist all the way to CLI and devtools for those comparisons.

----------

We'll have to update the error handling in devtools: https://github.com/ChromeDevTools/devtools-frontend/blob/28cf1d1a910a19f48c9a5d9d9374006f24179010/front_end/audits2/Audits2Panel.js#L708-L715  Can you make a note of that?

Reply:
yeah sg, done